bindings/go/oci/ctf/store.go

```diff
@@ -88,7 +90,7 @@ func (s *Store) ComponentVersionReference(ctx context.Context, component, versio
 type Repository struct {
 	archive ctf.CTF
 	repo    string
-	indexMu sync.RWMutex
+	indexMu *sync.RWMutex
```

Comment:
Are you sure about this? The mutex doesn't need to be a pointer if all the method receivers are pointers.

It is super rare that a mutex is a pointer and then it needs to be really explained as to why.

I see that you pass in the mutex from the Store. If I understand correctly, this means that the Repository CANNOT LIVE without a store. Because otherwise, this mutex will fail with a nil error since it's not initialized anywhere. And since it's an unexported field it means that this struct CANNOT BE USED outside of this package ever. Am I understanding this correctly?

Reply:
If yes, we might as well just make this an unexported struct.

Reply:
Now that it is private, let me ask this... do we even need it? Like, what would happen if the Store struct would be the one implementing all these methods?

If you say that it's there so it's a reasonable separation of concern, that's acceptable. :) WDYT?